app/models/concerns/publishable.rb

```diff
@@ -20,6 +20,10 @@ module Publishable
       where condition
     }
 
+    before_validation -> {
+      self.published_on ||= Time.zone.now if published?
```

Comment:
Changed to `Time.now` to avoid confusion.

BTW It returns time in system TZ, not UTC time (but here it does not matter anyway).